scripts/shared/lib/deploy_bundle

```diff
@@ -0,0 +1,249 @@
+# shellcheck shell=bash
+# shellcheck source=scripts/shared/lib/source_only
+. "${BASH_SOURCE%/*}"/source_only
+
+### Constants ###
+
+readonly CE_IPSEC_IKEPORT=500
+readonly CE_IPSEC_NATTPORT=4500
+readonly SUBM_IMAGE_REPO=localhost:5000
+readonly SUBM_IMAGE_TAG=${image_tag:-local}
+readonly MARKETPLACE_NAMESPACE="olm"
+readonly OPERATOR_NAME="submariner"
+readonly BROKER_NAMESPACE="submariner-k8s-broker"
+readonly BROKER_CLIENT_SA="submariner-k8s-broker-client"
+readonly INDEX_IMG=localhost:5000/submariner-operator-index:local
+readonly IPSEC_PSK="$(dd if=/dev/urandom count=64 bs=8 | LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1)"
+
+# Variables
+
+BROKER_K8S_API_SERVER=""
+BROKER_K8S_API_SERVER_TOKEN=""
+BROKER_K8S_CA=""
+
+### Functions ###
+
+function deploytool_prereqs() {
+  echo "VERSION:        ${VERSION}"
+  echo "SUBM_NS:        ${SUBM_NS}"
+  echo "CHANNEL:        ${CHANNEL}"
+  if [[ -z "${VERSION}" ]] ||
+   [[ -z "${SUBM_NS}" ]] ||
+   [[ -z "${CHANNEL}" ]]; then
+    echo "[ERROR] Required environment variables not loaded"
+    echo "    VERSION"
+    echo "    SUBM_NS"
+    echo "    CHANNEL"
+    exit 3
+  fi
+
+  # Create new namespace
+  run_subm_clusters create_submariner_namespace
+
+  # Create the custom catalog source
+  run_subm_clusters create_catalog_source
+
+  # Install the submariner operator bundle
+  run_subm_clusters install_bundle
+}
+
+function create_submariner_namespace {
+  echo "[INFO](${cluster}) Create the namespace..."
+  kubectl create namespace "${SUBM_NS}" || :
+}
+
+function create_catalog_source() {
+  echo "[INFO](${cluster}) Create the catalog source..."
+
+  kubectl delete catalogsource/submariner-catalog-source -n "${MARKETPLACE_NAMESPACE}" --wait --ignore-not-found
+
+  # Create the CatalogSource
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: submariner-catalog-source
+  namespace: ${MARKETPLACE_NAMESPACE}
+spec:
+  sourceType: grpc
+  image: ${INDEX_IMG}
+  displayName: Submariner Catalog Source
+  publisher: Submariner.io (Test)
+  updateStrategy:
+    registryPoll:
+      interval: 5m
+EOF
```

Comment:
This would be better if it was in a real YAML file which gets validated. You can use `render_template` function from `clusters.sh`